### PR TITLE
Add alternate retrieval of user's id in get-session

### LINKED ISSUE #1308: `/admin/auth` with api key returning 400 Bad Request
# Issue
Hi, when I try to get session with api key via `/admin/auth` I get 400 Bad Request.

# Cause
After digging a little I found that error appears here [get-session.ts#L25](https://github.com/medusajs/medusa/blob/master/packages/medusa/src/api/routes/admin/auth/get-session.ts#L25)
```ts
  try {
    const userService: UserService = req.scope.resolve("userService")
    const user = await userService.retrieve(req.user.userId) // <---- here, userId is undefined

    const cleanRes = _.omit(user, ["password_hash"])
    res.status(200).json({ user: cleanRes })
  } catch (err) {
    res.sendStatus(400)
  }
```
The problem is that when we send request with access_token the property `req.user.userId` is not set. Instead user id is in `req.user.id`.

# Solutions
I could PR solution but I don't know if my ideas isn't a little bit silly
We could just take from both properties like this
```ts
  try {
    const userService: UserService = req.scope.resolve("userService")
    const user = await userService.retrieve(req.user.userId || req.user.id)

    const cleanRes = _.omit(user, ["password_hash"])
    res.status(200).json({ user: cleanRes })
  } catch (err) {
    res.sendStatus(400)
  }
```
Or change here [passport.js#L52](https://github.com/medusajs/medusa/blob/master/packages/medusa/src/loaders/passport.js#L52) like this:
```js
  if (auth.success) {
    done(null, { ...auth.user, userId: auth.user.id })
  }
```

## Btw
Couldn't we log this `err` from try...catch? Coz currently it's failing, telling user it's their mistake and not showing any details at all...

### PR DESCRIPTION
**Pull request by gefgu**
### What: 
Fixes #1308  

### Why:
The solution was discussed, but it had no PR. So, I made one.

### CODE REVIEW

**Review by kuyazee — APPROVED** [review]

File: packages/medusa/src/api/routes/admin/auth/get-session.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ import _ from "lodash"
 export default async (req, res) => {
   try {
     const userService: UserService = req.scope.resolve("userService")
-    const user = await userService.retrieve(req.user.userId)
+    const user = await userService.retrieve(req.user.userId || req.user.id)
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/auth/get-session.ts:56]: todo: Should use the nullish coalescing operation
```suggestion
    const user = await userService.retrieve(req.user?.userId ?? req.user?.id)
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just a minor change 💪

File: packages/medusa/src/api/routes/admin/auth/get-session.ts (hunk 1 above)
Comment (gefgu) [reply packages/medusa/src/api/routes/admin/auth/get-session.ts]: Ok. Thanks for the feedback.

**Review by olivermrbl — APPROVED** [review]
Thx!

**Comment by riqwan** [comment]
Hey, thanks for the PR! Since v2 brought a lot of architectural changes, we will be closing this PR since it no longer applies to our new setup.

If you would still like to open this PR, please rebase with the v1 branch and re-request a review.

**Pull request closed without merging**